File: vue-next-admin/src/api/kubernetes/namespace.ts
import { Namespace } from 'kubernetes-types/core/v1';

import request from '/@/utils/request';

export function useNamespaceApi() {
	return {
		listNamespace: (param: Object) => {
			return request({
				url: '/namespaces',
				method: 'get',
				params: param,
			});
		},
		updateNamespace: (param: Object, namespace: Namespace) => {
			return request({
				url: '/namespaces',
				method: 'put',
				params: param,
				data: namespace,
			});
		},
		createNamespace: (param: Object, namespace: Namespace) => {
			return request({
				url: '/namespaces',
				method: 'post',
				params: param,
				data: namespace,
			});
		},
		deleteNamespace: (param: Object, namespaceName: string) => {
			return request({
				url: `/namespaces/${namespaceName}`,
				method: 'delete',
				params: param,
			});
		},
	};
}

File: server/apps/cloud/api/namespace.go
package api

import (
	"github.com/lbemi/lbemi/pkg/core"
	"github.com/lbemi/lbemi/pkg/rctx"

	v1 "k8s.io/api/core/v1"
)

func ListNamespace(rc *rctx.ReqCtx) {
	c := rc.Request.Request.Context()
	clusterName := rc.QueryCloud()
	pageParam := rc.GetPageQueryParam()
	name := rc.Query("name")
	label := rc.Query("label")
	rc.ResData = core.V1.Cluster(clusterName).Namespaces().List(c, pageParam, name, label)
}

func GetNamespace(rc *rctx.ReqCtx) {
	c := rc.Request.Request.Context()
	clusterName := rc.QueryCloud()
	name := rc.PathParam("name")
	rc.ResData = core.V1.Cluster(clusterName).Namespaces().Get(c, name)
}

func DeleteNamespace(rc *rctx.ReqCtx) {
	c := rc.Request.Request.Context()
	clusterName := rc.QueryCloud()
	name := rc.PathParam("name")
	core.V1.Cluster(clusterName).Namespaces().Delete(c, name)
}

func CreateNamespace(rc *rctx.ReqCtx) {
	c := rc.Request.Request.Context()
	clusterName := rc.QueryCloud()
	var namespace v1.Namespace
	rc.ShouldBind(&namespace)
	rc.ResData = core.V1.Cluster(clusterName).Namespaces().Create(c, &namespace)

}

func UpdateNamespace(rc *rctx.ReqCtx) {
	c := rc.Request.Request.Context()
	clusterName := rc.QueryCloud()
	var namespace v1.Namespace
	rc.ShouldBind(&namespace)
	rc.ResData = core.V1.Cluster(clusterName).Namespaces().Update(c, &namespace)
}
